irohad/main/server_runner.cpp

```diff
@@ -16,35 +16,48 @@
  */
 
 #include "main/server_runner.hpp"
-#include <grpc++/server.h>
-#include <grpc++/server_builder.h>
-#include <grpc++/server_context.h>
-#include "logger/logger.hpp"
 
-ServerRunner::ServerRunner(const std::string &address)
-    : serverAddress_(address) {}
+#include <boost/format.hpp>
 
-ServerRunner &ServerRunner::append(std::unique_ptr<grpc::Service> service) {
-  services_.emplace_back(std::move(service));
+const auto kPortBindError = "Cannot bind server to address %d";
+
+ServerRunner::ServerRunner(const std::string &address, bool reuse)
+    : serverAddress_(address) {
+  if (not reuse) {
+    builder_.AddChannelArgument(GRPC_ARG_ALLOW_REUSEPORT, 0);
+  }
+}
+
+ServerRunner &ServerRunner::append(std::shared_ptr<grpc::Service> service) {
+  builder_.RegisterService(service.get());
+  services_.push_back(service);
```

Comment:
May be it would be better to only push services here but register only in run(), what do you think?

Reply:
Yes. As @luckychess noted, method name `append` does not imply registering service.